patroni/validator.py

```diff
@@ -354,6 +354,27 @@ def __init__(self, schema: Dict[str, Any]) -> None:
         self._schema = schema
 
 
+class CaseOptions(object):
+    """Map how a dict of optional configurations should be validated."""
+
+    def __init__(self, schema: Dict['Optional', Any]):
+        """Create a :class:`StrictCase` object`.
+
+        :param schema: the optional arguments which may appear in the configuration.
+            Each option may or may not be provided, and will work with `:class:OneOf` or similar validations.
+
+        :Example:
+
+            OneOf("nofailover", "failover_priority"): CaseOptions({
+                Optional("nofailover"): bool,
+                Optional("failover_priority"): IntValidator(min=0, raise_assert=True),
+            })
```

Comment:
Although there are places in this file that don't use this (yet), it might be good to show a working example that can be validated by `doctest`.

```suggestion
            >>> s = Schema({OneOf("nofailover", "failover_priority"): CaseOptions({
            ...   Optional("nofailover"): bool,
            ...   Optional("failover_priority"): IntValidator(min=0, raise_assert=True),
            ... })})
            ...
            >>> s({'nofailover': True})
            []
            >>> s({'nofailover': True, 'failover_priority': 0})
            ["  Multiple of ('nofailover', 'failover_priority') provided"]
```